manim/utils/bezier.py

```diff
@@ -464,146 +951,408 @@ def match_interpolate(
     )
 
 
-def get_smooth_cubic_bezier_handle_points(
-    points: Point3D_Array,
-) -> tuple[BezierPoints, BezierPoints]:
-    points = np.asarray(points)
-    num_handles = len(points) - 1
-    dim = points.shape[1]
-    if num_handles < 1:
-        return np.zeros((0, dim)), np.zeros((0, dim))
-    # Must solve 2*num_handles equations to get the handles.
-    # l and u are the number of lower an upper diagonal rows
-    # in the matrix to solve.
-    l, u = 2, 1
-    # diag is a representation of the matrix in diagonal form
-    # See https://www.particleincell.com/2012/bezier-splines/
-    # for how to arrive at these equations
-    diag: MatrixMN = np.zeros((l + u + 1, 2 * num_handles))
-    diag[0, 1::2] = -1
-    diag[0, 2::2] = 1
-    diag[1, 0::2] = 2
-    diag[1, 1::2] = 1
-    diag[2, 1:-2:2] = -2
-    diag[3, 0:-3:2] = 1
-    # last
-    diag[2, -2] = -1
-    diag[1, -1] = 2
-    # This is the b as in Ax = b, where we are solving for x,
-    # and A is represented using diag.  However, think of entries
-    # to x and b as being points in space, not numbers
-    b: Point3D_Array = np.zeros((2 * num_handles, dim))
-    b[1::2] = 2 * points[1:]
-    b[0] = points[0]
-    b[-1] = points[-1]
-
-    def solve_func(b: ColVector) -> ColVector | MatrixMN:
-        return linalg.solve_banded((l, u), diag, b)  # type: ignore
-
-    use_closed_solve_function = is_closed(points)
-    if use_closed_solve_function:
-        # Get equations to relate first and last points
-        matrix = diag_to_matrix((l, u), diag)
-        # last row handles second derivative
-        matrix[-1, [0, 1, -2, -1]] = [2, -1, 1, -2]
-        # first row handles first derivative
-        matrix[0, :] = np.zeros(matrix.shape[1])
-        matrix[0, [0, -1]] = [1, 1]
-        b[0] = 2 * points[0]
-        b[-1] = np.zeros(dim)
-
-        def closed_curve_solve_func(b: ColVector) -> ColVector | MatrixMN:
-            return linalg.solve(matrix, b)  # type: ignore
-
-    handle_pairs = np.zeros((2 * num_handles, dim))
-    for i in range(dim):
-        if use_closed_solve_function:
-            handle_pairs[:, i] = closed_curve_solve_func(b[:, i])
-        else:
-            handle_pairs[:, i] = solve_func(b[:, i])
-    return handle_pairs[0::2], handle_pairs[1::2]
+# Figuring out which Bezier curves most smoothly connect a sequence of points
```

Comment:
Figure out where comment was or just remove it if unnecessary

Reply:
Discussed in Discord, but I clarify for readers: in the end we dismissed this.
The confusion appeared because the diff was ruined.
But the comment is where it should belong.